src/htr2hpc/train/data.py

```diff
@@ -158,15 +169,32 @@ def get_training_data(api, output_dir, document_id, part_ids=None):
 
     # get segmentation data for each part of the document that is requested
     segmentation_data = [
-        get_segmentation_data(api, document_details, part_id, output_dir)
+        get_segmentation_data(
+            api, document_details, part_id, output_dir, transcription_id
+        )
         for part_id in part_ids
     ]
-    # if we're generating alto-xml (i.e., segmentation training data),
-    # serialize each of the parts we downloaded
-    [serialize_segmentation(seg, part) for (seg, part) in segmentation_data]
 
-    # NOTE: binary compiled data is only supported train and not segtrain
-    # compiled_data = compile_data(segmentations, output_dir)
+    # if transcription id is specified, compile as binary dataset
+    # for recognition training
+    if transcription_id:
+        segmentations = [seg for seg, _ in segmentation_data]
+        # NOTE: get code errors in kraken if the image path is not valid.
+        # Image path on created segments should be relative to current
+        # working directory. Must resolve so the kraken binary compile
+        # function can load image files by path.
+        build_binary_dataset(
+            segmentations,
+            output_file=str(output_dir / "train.arrow"),
+            num_workers=4,
```

Comment:
4 probably works fine for `num_workers` for now, but we might revisit this. I suppose this part of the code is being run on Della but not as the submitted slurm job, so we're working with what resources are available there.

Reply:
Yes, this is running on della outside of slurm. We can probably use more than 4 but wasn't sure if we wanted to use the same number as we use for the slurm job (which is configurable via command line args).